### PR TITLE
Add FAQ page to docs

### PR DESCRIPTION
**Pull request by simonbray**
One of the Planemo paper reviewers suggested adding a FAQ page to the Planemo documentation. Here's my attempt.

Please feel free to edit or add new sections (or, to suggest new questions and I'll try and write answers to them.)

### CODE REVIEW

File: docs/faqs.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+=============================
+
+Planemo is a command-line application which is primarily used for developing Galaxy and CWL tools,
+workflows and training materials. It's also used to deploying the tools and workflows once developed
+(tools to the Galaxy ToolShed; workflows to the Intergalactic Workflow Commission) and executing
+Galaxy-based analyses via the command-line, if you prefer or need to use that rather than Galaxy's
+graphical interface.
+
+Is Planemo the right tool for me?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:17]:
```suggestion
Is Planemo the right tool for me?
=================================
```

File: docs/faqs.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Yes, if you are comfortable using the command-line and one of the following is true:
+  - You want to use a piece of scientific software not currently available in Galaxy and therefore need to wrap it yourself.
+  - You want to develop CWL tools or workflows.
+  - You're writing a tutorial for the `Galaxy Training Network`_ and would like to use a pre-prepared template.
+  - You want to run Galaxy workflows from the command line.
+
+What is the difference between Galaxy and CWL? Which should I use?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:26]:
```suggestion
What is the difference between Galaxy and CWL? Which should I use?
==================================================================
```

File: docs/faqs.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+CWL (Common Workflow Language) is a workflow specification, rather than a workflow management system.
+It aims to allow the description of scientific pipelines in a way that is independent of the particular
+workflow manager used; if this is important to you, using CWL would be an excellent idea. CWL workflows
+are supported by a variety of workflow managers, including Toil, cwltool, Arvados and Galaxy. 
+
+Both Galaxy and CWL place a high value on community, open-source code and FAIR data analysis.
+
+Can I develop Galaxy tools without Planemo?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:42]:
```suggestion
Can I develop Galaxy tools without Planemo?
===========================================
```

File: docs/faqs.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+Yes, you can, but we highly discourage you from doing so. Planemo encourages best practices in
+software development and follows guidelines agreed on by the Galaxy community, particularly
+test-driven development.
+
+How can I learn tool development?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:49]:
```suggestion
Where can I learn to write tools?
=================================
```

File: docs/faqs.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+We have a `tutorial`_ in this documentation and another incorporated into the
+Galaxy Training Network, which should get you off to a good start.
+
+How do I set up deployment to the ToolShed with Planemo?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:55]:
```suggestion
How do I set up deployment to the ToolShed with Planemo?
========================================================
```

File: docs/faqs.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+We recommend using continuous integration (e.g. GitHub Actions) to achieve this and provide
+a `template`_ which you can use to set up a GitHub repository which automatically deploys tools
+to the ToolShed when a pull request is merged.
+
+If you don't want to handle tool deployment yourself, you can also submit your tool wrappers to
+a community repository such as the `IUC`_. This also has the advantage that your wrappers will be
+reviewed by experienced Galaxy users and developers.
+
+How can I automate workflow execution with Planemo?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:66]:
```suggestion
How can I automate workflow execution with Planemo?
===================================================
```

File: docs/faqs.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+Have a look at this `section`_ of the documentation, or this `training`_ provided by the Galaxy
+Training Network.
+
+How can I contribute to the project?
+=============================
```
Comment (nsoranzo) [inline docs/faqs.rst:72]:
```suggestion
How can I contribute to the project?
====================================
```

File: docs/faqs.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+We would love to see new contributions to Planemo! Probably the easiest way to contribute is
+to this documentation; if you find any material which is unclear, out-of-date, or simply wrong,
+please `make a GitHub issue`_ or a pull request to our `GitHub repository`_. If you are comfortable
+writing code, have a look at our `list of issues`_ for ideas.
```
Comment (nsoranzo) [inline docs/faqs.rst:77]: Maybe simply refer to https://github.com/galaxyproject/planemo/blob/master/CONTRIBUTING.rst ?

File: docs/faqs.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+.. _tutorial: https://planemo.readthedocs.io/en/latest/writing_standalone.html
+.. _Galaxy Training Network: https://training.galaxyproject.org/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html
```
Comment (nsoranzo) [inline docs/faqs.rst:80]:
```suggestion
.. _Galaxy Training Network: https://training.galaxyproject.org/
```

File: docs/faqs.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+=============================
+
+Yes, you can, but we highly discourage you from doing so. Planemo encourages best practices in
+software development and follows guidelines agreed on by the Galaxy community, particularly
+test-driven development.
+
+How can I learn tool development?
+=============================
+
+We have a `tutorial`_ in this documentation and another incorporated into the
```
Comment (nsoranzo) [inline docs/faqs.rst:51]:
```suggestion
We have a `tutorial`_ in this documentation and `another <https://training.galaxyproject.org/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html>`__ incorporated into the
```

File: docs/faqs.rst [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+to this documentation; if you find any material which is unclear, out-of-date, or simply wrong,
+please `make a GitHub issue`_ or a pull request to our `GitHub repository`_. If you are comfortable
+writing code, have a look at our `list of issues`_ for ideas.
+
+.. _tutorial: https://planemo.readthedocs.io/en/latest/writing_standalone.html
+.. _Galaxy Training Network: https://training.galaxyproject.org/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html
+.. _template: https://github.com/galaxyproject/galaxy-tool-repository-template
+.. _IUC: https://github.com/galaxyproject/tools-iuc
+.. _section: https://planemo.readthedocs.io/en/latest/running.html
+.. _Galaxy Training Network: https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html
```
Comment (nsoranzo) [inline docs/faqs.rst:84]:
```suggestion
.. _training: https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html
```

File: docs/faqs.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+(tools to the Galaxy ToolShed; workflows to the Intergalactic Workflow Commission) and executing
+Galaxy-based analyses via the command-line, if you prefer or need to use that rather than Galaxy's
+graphical interface.
+
+Is Planemo the right tool for me?
+=================================
+
+Yes, if you are comfortable using the command-line and one of the following is true:
+  - You want to use a piece of scientific software not currently available in Galaxy and therefore need to wrap it yourself.
+  - You want to develop CWL tools or workflows.
```
Comment (bgruening) [inline docs/faqs.rst:21]: Should we add here Galaxy tools as well?

File: docs/faqs.rst (hunk 8 above)
Comment (simonbray) [reply docs/faqs.rst]: Yes, thank you!

File: docs/faqs.rst [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+are supported by a variety of workflow managers, including Toil, cwltool, Arvados and Galaxy. 
+
+Both Galaxy and CWL place a high value on community, open-source code and FAIR data analysis.
+
+Can I develop Galaxy tools without Planemo?
+===========================================
+
+Yes, you can, but we highly discourage you from doing so. Planemo encourages best practices in
+software development and follows guidelines agreed on by the Galaxy community, particularly
+test-driven development.
```
Comment (bgruening) [inline docs/faqs.rst:46]: Do we want to recommend here the Galaxy language server as well?

File: docs/faqs.rst (hunk 12 above)
Comment (bgruening) [reply docs/faqs.rst:21]: Or maybe add a new bullet point about developing Galaxy workflows?

Comment (simonbray) [reply docs/faqs.rst:22]: > Should we add here Galaxy tools as well?

That's meant to be the first point.

File: docs/faqs.rst [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+It aims to allow the description of scientific pipelines in a way that is independent of the particular
+workflow manager used; if this is important to you, using CWL would be an excellent idea. CWL workflows
+are supported by a variety of workflow managers, including Toil, cwltool, Arvados and Galaxy. 
+
+Both Galaxy and CWL place a high value on community, open-source code and FAIR data analysis.
+
+Can I develop Galaxy tools without Planemo?
+===========================================
+
+Yes, you can, but we highly discourage you from doing so. Planemo encourages best practices in
```
Comment (mvdbeek) [inline docs/faqs.rst:45]:
```suggestion
Yes, you can, but it is more complicated to do so and will take more time. Planemo encourages best practices in
```
?

**Review by bernt-matthias — APPROVED** [review]

File: docs/faqs.rst [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+If you use the Visual Studio Code editor, we recommend using Planemo in conjunction with the
+`Galaxy Language Server`_, which encourages development best practices and should greatly
+improve the speed at which you can develop new Galaxy tools.
+
+How do I set up deployment to the ToolShed with Planemo?
+========================================================
+
+We recommend using continuous integration (e.g. GitHub Actions) to achieve this and provide
+a `template`_ which you can use to set up a GitHub repository which automatically deploys tools
```
Comment (nsoranzo) [inline docs/faqs.rst:63]:
```suggestion
a `template repository`_ which you can use to set up a GitHub repository which automatically deploys tools
```

File: docs/faqs.rst [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+====================================
+
+We would love to see new contributions to Planemo! Please have a look `here`_ to see the different
+ways you can help.
+
+
+.. _tutorial: https://planemo.readthedocs.io/en/latest/writing_standalone.html
+.. _Galaxy Training Network: https://training.galaxyproject.org/
+.. _Galaxy Language Server: https://github.com/galaxyproject/galaxy-language-server
+.. _template: https://github.com/galaxyproject/galaxy-tool-repository-template
```
Comment (nsoranzo) [inline docs/faqs.rst:86]:
```suggestion
.. _template repository: https://github.com/galaxyproject/galaxy-tool-repository-template
```

File: docs/faqs.rst [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+
+How do I set up deployment to the ToolShed with Planemo?
+========================================================
+
+We recommend using continuous integration (e.g. GitHub Actions) to achieve this and provide
+a `template`_ which you can use to set up a GitHub repository which automatically deploys tools
+to the ToolShed when a pull request is merged.
+
+If you don't want to handle tool deployment yourself, you can also submit your tool wrappers to
+a community repository such as the `IUC`_. This also has the advantage that your wrappers will be
```
Comment (nsoranzo) [inline docs/faqs.rst:67]:
```suggestion
a community repository such as the `IUC's one`_. This also has the advantage that your wrappers will be
```

File: docs/faqs.rst [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+We would love to see new contributions to Planemo! Please have a look `here`_ to see the different
+ways you can help.
+
+
+.. _tutorial: https://planemo.readthedocs.io/en/latest/writing_standalone.html
+.. _Galaxy Training Network: https://training.galaxyproject.org/
+.. _Galaxy Language Server: https://github.com/galaxyproject/galaxy-language-server
+.. _template: https://github.com/galaxyproject/galaxy-tool-repository-template
+.. _IUC: https://github.com/galaxyproject/tools-iuc
```
Comment (nsoranzo) [inline docs/faqs.rst:87]:
```suggestion
.. _IUC's one: https://github.com/galaxyproject/tools-iuc
```

**Pull request merged**